spec/jobs/old_keys_reminder_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe Jobs::OldKeysReminder do
+  let!(:google_secret) { SiteSetting.create!(name: 'google_oauth2_client_secret', value: '123', data_type: 1) }
+  let!(:instagram_secret) { SiteSetting.create!(name: 'instagram_consumer_secret', value: '123', data_type: 1) }
+  let!(:api_key) { Fabricate(:api_key, description: 'api key description') }
+  let!(:admin) { Fabricate(:admin) }
+
+  it "detects old keys" do
+    SiteSetting.notify_about_secrets_older_than = "2 years"
+    freeze_time 2.years.from_now
+    expect(described_class.new.send(:old_site_settings_keys).sort).to eq([google_secret, instagram_secret].sort)
```

Comment:
I don't think we should be testing private methods. A test should stick to the public API surface because this prevents us from refactoring in the future.

In this case I would recommend moving the `old_xxx` methods to `SiteSetting` as public and testing using those.

For the `title`, `body` etc you can look for the post that was created after the job finishes.